File: packages/python.org/lodash/tests/number.test.py
"""
object
"""

import unittest
import libs as _


class TestNumberMethods(unittest.TestCase):
    def test_clamp(self):
        self.assertEqual(_.clamp(-10, -5, 5), -5)
        self.assertEqual(_.clamp(10, -5, 5), 5)

    def test_in_range(self):
        self.assertTrue(_.in_range(3, 2, 4))
        self.assertTrue(_.in_range(4, 8))
        self.assertFalse(_.in_range(4, 2))
        self.assertFalse(_.in_range(2, 2))
        self.assertTrue(_.in_range(1.2, 2))
        self.assertFalse(_.in_range(5.4, 4))
        self.assertFalse(_.in_range(-3, -2, -6))

    def test_random(self):
        random1 = _.random(5)
        self.assertTrue(0 <= random1 and random1 <= 5)
        random2 = _.random(5, 10)
        self.assertTrue(5 <= random2 and random2 <= 10)


if __name__ == "__main__":
    unittest.main()

File: packages/go.dev/libraries/utils/lodash/number/number_test.go
package number

import (
	"fmt"
	"testing"
)

func TestClamp(t *testing.T) {
	// clamp
	var clamped1 float64 = Clamp(-10, -5, 5)
	var clamped2 float64 = Clamp(10, -5, 5)
	if clamped1 == -5 && clamped2 == 5 {
		fmt.Println("clamped true")
	}
	// inRange

}

func TestInRange(t *testing.T) {
	var inRange1 bool = InRange(3, 2, 4)
	var inRange2 bool = InRange(4, 0, 8)
	var inRange3 bool = InRange(4, 0, 2)
	var inRange4 bool = InRange(2, 0, 2)
	if inRange1 && inRange2 && !inRange3 && !inRange4 {
		fmt.Println("isRange true")
	}
}

func TestRandom(t *testing.T) {
	var random1 = Random(0, 5)
	var random2 = Random(5, 10)
	if 0 < random1 && random1 < 5 && 5 < random2 && random2 < 10 {
		fmt.Println("random true")
	}
}
